int32[7] rack_pos
int32[7] target_row
int32[7] target_col
